### PR TITLE
Color Schemes: Add postcss custom properties to main style build step 

### PR DESCRIPTION
**Pull request by frontdevde**
#### This PR 
- is part of a series of PRs aiming to provide support for Color Schemes in Calypso.

-  focuses on adding the postcss plugin `postcss-custom-properties` to the main style build step

-  adds `postcss.config.json` as a central place for postcss plugin configurations

- is a continuation of PR https://github.com/Automattic/wp-calypso/pull/17366, which will be closed in favor of this and a few other smaller PRs.

#### How to test

Run the CSS build and ensure that the output is identical to the output generated prior to this PR.

#### Notes

~~This PR currently compares its changes to PR https://github.com/Automattic/wp-calypso/pull/18905. This is so the changes can be easily identified. This PR needs to be pointed at master once PR https://github.com/Automattic/wp-calypso/pull/18905 has been merged.~~
(#18905 was merged & this is now targeting `master`)

For a more detailed description of the feature, the chosen approach and its advantages feel free to check out my post on the subject: p4TIVU-75d-p2 (internal reference).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -173,13 +173,14 @@
     "analyze-bundles": "webpack-bundle-analyzer stats.json public -h 0.0.0.0 -p 9898",
     "analyze-css": "node bin/analyze-css.js",
     "autoprefixer": "postcss -r --use autoprefixer",
+    "postcss": "postcss -r -u postcss-custom-properties -u autoprefixer -c postcss.config.json",
```
Comment (jblz) [inline package.json]: It feels a bit weird to leave `autoprefixer` split out, but have `custom-properties` step to be part of a wrapping task.  Should we do one of:
* also split out `custom-properties`
* bring `autoprefixer` into the `postcss` task

...? Is there any reason we'd want to run either of those two things independently?

File: postcss.config.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "postcss-custom-properties": {
+        "preserve": "true",
+        "warnings": "false"
+    }
+}
```
Comment (jblz) [inline postcss.config.json]: Please include a newline at the end of the file.

I'm not seeing it in the coding guidelines, but it's set in the [`.editorconfig`](https://github.com/Automattic/wp-calypso/blob/110106d322fd8138e9f9d203ac46a0518df2fd82/.editorconfig#L15) & is generally a good practice.

i.e. when you do something like `cat postcss.config.json`, your prompt starts w/ a clean line, etc.

File: package.json (hunk 1 above)
Comment (frontdevde) [reply package.json]: Good point! The reason for the current state is that we'd only want to use an additional postcss step for the parts of the build that actually need it. Currently, we only make use of css custom properties in the main styles and not the sections. Thus the idea is to use it only for the build step of the main styles and don't increase the build time where it's not necessary.

File: postcss.config.json (hunk 2 above)
Comment (jblz) [reply postcss.config.json]: Spelled this out in #18927 :)

Comment (frontdevde) [reply postcss.config.json]: Newline should be there now. Pushed the change ~40min ago but GH is a bit slow today. 
The newline is a default setting in our `prettier` config btw. So all new files should have it by default. So used to it being added on save. Just had temporarily deactivated the option in my IDE that runs `prettier` on save. 🙈

**Comment by jblz** [comment]
This is working as advertised & the rationale behind the npm script status makes sense to me. 👍

**Pull request merged**